examples/CMakeLists.txt

```
@@ -2,6 +2,11 @@
 
 set(ALL_EXAMPLES scope_example unique_resource unique_resource-file)
 
+# module tests will only compile with gcc15 or clang20 and above
+if(CMAKE_CXX_SCAN_FOR_MODULES AND CMAKE_CXX_STANDARD GREATER 20)
```

```suggestion
if(CMAKE_CXX_SCAN_FOR_MODULES)
```

any checking of C++20 mode is redundant as the c++ side enforces that we're using c++20 and above.  In general I don't think we should really ever reference the CXX standard in cmake files (even to enforce the minimum) and should always do in library headers since if someone takes the header and drops it into the project our cmake stuff isn't going to help them.

---

std::print is c++23, without import std; the module example will fail!

---

I suspect this comment was meant to go on the scope-module.cpp so I'll discuss there

---

done